vassal-app/src/main/java/VASSAL/tools/filechooser/FileChooser.java

```diff
@@ -70,49 +65,156 @@ public static FileChooser createFileChooser(Component parent, DirectoryConfigure
    *          A FileConfigure that stores the preferred starting directory of the FileChooser in preferences
    */
   public static FileChooser createFileChooser(Component parent, DirectoryConfigurer prefs, int mode) {
-    FileChooser fc;
-    if (SystemUtils.IS_OS_MAC_OSX) {
-      // Mac has a good native file dialog
-      fc = new NativeFileChooser(parent, prefs, mode);
-    }
-    else if (mode == FILES_ONLY && SystemUtils.IS_OS_WINDOWS) {
-      // Window has a good native file dialog, but it doesn't support selecting directories
-      fc = new NativeFileChooser(parent, prefs, mode);
```

Comment:
Have you checked that the file choosers this switches us to support selecting directories?

Reply:
Checked and changed this PR to use the native dialog only for file load/save on Linux, while keeping the Swing dialog for selecting directories.